gooddata-sdk/gooddata_sdk/table.py

```diff
@@ -258,7 +278,188 @@ def _create_dimension(bucket: VisualizationBucket, measures_item_identifier: Opt
+    direction: SortDirection
+    measure_dim_identifier: str
+    data_column_locators: list[Locator]
+
+    def to_dict(self) -> dict:
+        locator_tuples = [locator.to_kv() for locator in self.data_column_locators]
+        return {
+            "value": {
+                "dataColumnLocators": {self.measure_dim_identifier: {k: v for k, v in locator_tuples}},
+                "direction": _SORT_DIRECTION_TO_API[self.direction],
+            }
+        }
+
+
+SortKey = Union[SortKeyAttribute, SortKeyValue]
+
+
+def _create_data_col_locators(locators: list[VisualizationSortLocator]) -> list[Locator]:
+    converted_locators: list[Locator] = []
+    for locator in locators:
+        if locator.type == LocatorItemType.ATTRIBUTE:
+            converted_locators.append(
+                AttributeLocator(
+                    attribute_identifier=locator.locator["attributeIdentifier"],
+                    element=locator.locator["element"],
+                )
+            )
+        if locator.type == LocatorItemType.MEASURE:
+            converted_locators.append(
+                MeasureLocator(
+                    measure_identifier=locator.locator["measureIdentifier"],
+                )
+            )
+    return converted_locators
+
+
+def _create_dims_with_sorts(dims: list[TableDimension], sorts: list[VisualizationSort]) -> list[TableDimension]:
+    """
+    Places sorting into dimensions. Returns the same dimensions objects but with modified sorting.
+
+    Tiger does sorting differently from bear so this is somewhat more complicated than pure object conversions.
+
+    1. Sorting is now placed in the dimension that has to be sorted
+    2. When sorting by attribute (headers), then the attribute sort key must be placed into the dimension that
+       contains the attribute
+    3. When sorting by measure, we now fall back to 'bear-like' behavior: the dimension opposite to the one
+       that contains the measures will be sorted. It will be sorted using the measure (possibly scoped for
+       particular attribute values) located in the MeasureGroup dimension.
+
+    At the end, this function walks the sort items in the order defined by the user and distributes them into
+    the dimensions. The function is lenient for now and will log warnings and ignore anything weird that it
+    cannot process (e.g. measure sorting when there is no dim with MeasureGroup, or if there is only dim with
+    MeasureGroup and no other dim).
+
+    @param dims - dimensions to add sorting to
+    @param sorts - sort items defined by SDK user
+    """
+    if not sorts:
+        return dims
+
+    non_measure_dim_idx = _get_dim_idx_for_predicate(dims, lambda x: _MEASURE_GROUP_IDENTIFIER not in x.item_ids)
+    measure_dim_idx = _get_dim_idx_for_predicate(dims, lambda x: _MEASURE_GROUP_IDENTIFIER in x.item_ids)
+    measure_dim = dims[measure_dim_idx] if measure_dim_idx else None
+
+    sorting: list[list[SortKey]] = [[] for _ in dims]
```

Comment:
`sorting: list[list[SortKey]] = [[] * len(dims)]`

Reply:
Hmm, seems like this produces just a list with a single list for any value of len(dims)>1. We need list of multiple empty lists here (in case of len(dims)>1). So I would keep it as is if you agree.